src/modules/image/producer/image_producer.cpp

```diff
@@ -141,72 +124,14 @@ spl::shared_ptr<core::frame_producer> create_producer(const core::frame_producer
     }
 
     auto length = get_param(L"LENGTH", params, std::numeric_limits<uint32_t>::max());
+    auto scale_mode = core::scale_mode_from_string(get_param(L"SCALE_MODE", params, L"STRETCH"));
```

Comment:
@Julusian which file should I put it in?

Reply:
I would guess the same file as `get_param` is defined in?
but it looks like that may be messy, which is either a good argument for a new amcp parsing helpers file, or to just not do it

Reply:
@Julusian I thought about that, but `src/common/param.h` where it's defined seems too generic for that. And, since it's only used in 2 places, I've opted to leave it alone.